src/main/java/dk/cngroup/trainings/spring/springassignment/controller/rest/CareTakerController.java

```diff
@@ -39,70 +38,80 @@ public CareTakerController(CareTakerService careTakerService) {
 			return ResponseEntity.notFound().build();
+		} catch (InvalidCareTakerException e) {
+			return ResponseEntity.badRequest().build();
 		}
 	}
 
-
 	@RequestMapping(value = "/{id}", method = RequestMethod.GET)
 	public ResponseEntity<CareTaker> getCareTakerById(@PathVariable("id") long id) {
-		if (careTakerService.getCareTakerById(id).isPresent()) {
+		try {
 			return new ResponseEntity<>(careTakerService.getCareTakerById(id).get()
 					, HttpStatus.OK);
-		} else {
+		} catch (CareTakerNotFoundException e) {
 			return ResponseEntity.notFound().build();
 		}
 	}
 
 	@RequestMapping(path = "{id}", method = RequestMethod.DELETE)
-	public ResponseEntity<String> deleteCareTaker(@PathVariable("id") long id) {
-		if (careTakerService.deleteCareTakerById(id)) {
+	public ResponseEntity<String> deleteCareTakerById(@PathVariable("id") long id) {
+		try {
+			careTakerService.deleteCareTakerById(id);
 			return new ResponseEntity<>("Success", HttpStatus.OK);
-		} else {
-			return new ResponseEntity<>("Fail: Id doesn't exist", HttpStatus.NOT_FOUND);
+		} catch (CareTakerNotFoundException e) {
+			return new ResponseEntity<>("Fail: Care Taker with id " + id + " doesn't exist"
+					, HttpStatus.NOT_FOUND);
 		}
 	}
 
 	@RequestMapping(path = "/{id}/animals", method = RequestMethod.POST)
 	public ResponseEntity<Animal> addNewAnimalToExistingCareTaker(@PathVariable("id") long id,
 																  @RequestBody @Valid Animal animal) {
-		if (careTakerService.getCareTakerById(id).isPresent()) {
-			Optional<Animal> addedAnimal;
-			try {
-				addedAnimal = careTakerService.addNewAnimalToExistingCareTaker(id, animal);
-				return new ResponseEntity<>(addedAnimal.get(), HttpStatus.OK);
-			} catch (InvalidAnimalException e) {
-				return ResponseEntity.badRequest().build();
-			}
-		} else {
+		Optional<Animal> addedAnimal;
+		try {
+			addedAnimal = careTakerService.addNewAnimalToExistingCareTaker(id, animal);
+			return new ResponseEntity<>(addedAnimal.get(), HttpStatus.OK);
+		} catch (InvalidAnimalException e) {
+			return ResponseEntity.badRequest().build();
+		} catch (CareTakerNotFoundException e) {
 			return ResponseEntity.notFound().build();
 		}
 	}
 
+
 	@RequestMapping(value = "/{id}/animals", method = RequestMethod.GET)
 	public ResponseEntity<List<Animal>> getAnimalsInCareByCareTakerId(@PathVariable("id") long id) {
-		Optional<CareTaker> careTakerById = careTakerService.getCareTakerById(id);
-		if (careTakerById.isPresent()) {
+		Optional<CareTaker> careTakerById;
+		try {
+			careTakerById = careTakerService.getCareTakerById(id);
 			return new ResponseEntity<>(careTakerById.get().getAnimals()
 					, HttpStatus.OK);
-		} else {
+		} catch (CareTakerNotFoundException e) {
 			return ResponseEntity.notFound().build();
 		}
 	}
 
 	@RequestMapping(path = "/{careTakerId}/animals/{animalId}", method = RequestMethod.PUT)
 	public ResponseEntity<String> addExistingAnimalToExistingCareTaker(@PathVariable("careTakerId") long careTakerId,
```

Comment:
Same here, you can return CareTaker or Animal

Reply:
same remark for me as well.